### ISSUE TITLE
"Monster" KORD Map

### ISSUE THREAD

**Issue by siyuanr** [zlsa/atc#387]
@tedrek I saw you wanting to make a difficult Chicago O'HARE map in #284, and I just went ahead and have made a draft [in my fork](https://siyuanr.github.io/atc/). All the necessary waypoints are there, and I am going to add the STAR routes later.

<img width="956" alt="screen shot 2016-02-11 at 06 53 31" src="https://cloud.githubusercontent.com/assets/15655087/12970222/402620e4-d08c-11e5-97a4-a185ffa61571.png">

**Comment by Maverick283** [comment]
Wow this looks... Challenging! This is gonna need a huge amount of traffic all at once :D

**Comment by siyuanr** [comment]
Mostly done. All waypoints, STARs, departures angles are in place. A few remarks:
- What I need now is to adjust the type and frequency of all departures and arrivals. I also need a weighted list of airlines for all departures and arrivals; @tedrek @Maverick283 @erikquinn @Fechulo could you please help?
- Some STARs partially overlap or go close with each other. In such cases the inbound altitudes are different in order to maintain separation;
- `O'HARE ONE` departure stipulates that all planes be given direct radar vector to first NAVAID fix. Since in our game the handoff occurs not at fixes but at exit angles, I computed the exit angles as the angles between map center and handoff fix;
- `OXI6` STAR asks for 340 heading after `STYLE`. `STYLE` is outside the map; despite the spawn radial can be adjusted to direct planes inbound, the first plane on this STAR sometimes is spawned too far away from the radial and cannot head inbound. I have to add an unrealistic fix of `ORD` as a result;
- Wind direction is pretty diverse throughout the year. Given that variable wind is yet to be implemented, I took the "most dominant" direction (SSW, about 10.1% of the days).

**Comment by Maverick283** [comment]
While this is a [list of airlines](http://www.flychicago.com/OHare/EN/MyFlight/Airline/Airline-Information.aspx) flying in and out of O'Hare ([same in wikipedia but with destinations](https://en.wikipedia.org/wiki/O%27Hare_International_Airport#Airlines_and_destinations)) this is what I got using [this table](http://www.flightradar24.com/data/airports/ORD) (Opened on 02/11/2016 15:45 UTC) and some excel magic...

![image](https://cloud.githubusercontent.com/assets/12700191/12982712/442dd3bc-d0e6-11e5-8738-9b766932d684.png)

Note how the sum does not add up, that is because I removed the bottom ones...

This data is only somewhat representing the actual takeoffs and landings of the airlines as it is a snapshot with the range of only one day.

The full spreadsheet can be viewed [here](http://1drv.ms/1Q9F5Rz). If you copy and paste a new list in the first couple columns, the list on the side will automatically update and list all entries of the list you pasted. To get a list, just go [here](http://www.flightradar24.com/data/airports/ORD), and copy paste one of the two lists below the map. Should work for any airport btw. If oyu know what you are doing in excel, feel free to unlock the sheets, the password is "GithubATC". If you don't know what you're doing, for your own sake keep it locked, that way you can't mess up anything ;)

**Comment by erikquinn** [comment]
^That should cover your needs, but for future reference, I actually highly recommend flightstats. They'll suppress codeshares, which I didn't see a way to do on the flightradar24 page. Between flightstats and flightaware, you really have everything you need.
[http://www.flightstats.com/go/FlightStatus/flightStatusByAirport.do?airportQueryType=0](http://www.flightstats.com/go/FlightStatus/flightStatusByAirport.do?airportQueryType=0)
[http://flightaware.com/](http://flightaware.com/)
[http://flightaware.com/statistics/ifr-route/](http://flightaware.com/statistics/ifr-route/)

**Comment by Maverick283** [comment]
Ohhhh Codeshare.... Didn't think about that... ^^ 
I don't really find the key information on those pages, it just a list with deps and arrs and you'd have to count them by hand or am I missing something? As I didn't want to count them by hand and thought this problem will occur more often I tried to figure out a way to get this data by a program (in this case: excel). I [edited the excel](http://1drv.ms/1Q9F5Rz) so that it too suppresses codeshare flights. It gets a lot more boring though, as there is pretty much only United flying in and out...

**Comment by siyuanr** [comment]
I use STATA to manipulate the departure data on Feb. 11, 2016. To suppress codeshare flights I simply drop those entries with "^" (then tabulate the "Carrier" data entries):

``` Stata
drop if strpos("Flight", "^")!=0
tab Carrier
```

The results are astonishing. Clean departure data is on the left, while the codeshare tainted data in on the right. It goes to show how much of a presence British Airways has built using codeshares:

<img width="600" alt="screen shot 2016-02-12 at 16 06 43" src="https://cloud.githubusercontent.com/assets/15655087/13010663/35e3a684-d1a3-11e5-8292-af06d9168226.png">

I plan to use the percentage number as weight.

A big problem encountered during the mapping process is the airlines and aircraft models. I am weighing between submitting a lump-sum pull request with maps, airlines and aircraft changes, or filing them separately:
- Some airlines are missing (e.g. Qatari, ANA);
- Some aircraft models are missing, notably the A350 XWB series and other variations of A320 (the neos), B737, and also B777-200LR (not ER);
- Some airlines do not differentiate between `long` and `default` fleets. It would be a little strange if we see a Lufthansa A320 flying into O'hare...

**Comment by siyuanr** [comment]
@Maverick283 I think codeshare is a lesser cause for the dominance of Untied (and American, Delta for that matter) at `KORD`. The problem is that FlightRadar24 does not distinguish between those actually run by United and those run by the minor companies (ExpressJet, GoJet, etc.). FlightAware does distinguish between these flights though, but they stubbornly displays only 40 flights per page, so copying the amount of 24 hours is a huge project.

**Comment by sixeyeco** [comment]
Excel's <b>Data/From Web</b> feature would make this easier, particularly since flightaware uses predictable url's. 

To get the fisrt 20: https://flightaware.com/live/airport/KSFO
To get the next 20: http://flightaware.com/live/airport/KSFO/arrivals?;offset=20;order=actualarrivaltime;sort=DESC
Edit the url to add another 20 to "offset=20"

I did the first 140 or so for KSFO but you can edit it to any airport on flightware. 

Open this workbook, hit CTRL+ALT+F5. Then, filter out blanks and "type" in column B for a clean list.  

[Flight-data.xlsx](https://github.com/zlsa/atc/files/129150/Flight-data.xlsx)

**Comment by siyuanr** [comment]
@sixeyeco Thanks for the excel file! I had to change it a little bit because in Germany (where I am) the default website you see is different than the one you would in the US.

I decided to put #382 on hold (waiting for #324 anyway), delete the branch and start anew. The `KORD` pull request would incorporate the airport map and any new or edited airlines going in and out of the airport. I've completed all the statistical work (see Excel file attached). A few remarks before starting:
- Data is from arrival and departure on Friday Jan. 12;
- Departure weight among airlines is not the actual departure frequency, but the combined frequency (total presence) of both departure and arrival, taking into account of the non-daily connections;
- **Initial goal is 60 departures and roughly 60 arrivals per hour;**
- STARs OXI6, JVL7, and ESSPO2 are not regularly assigned; their airline weight follows the total presence data; their frequency is 1 per hour.

[KORD.xlsx](https://github.com/zlsa/atc/files/129533/KORD.xlsx)
